File: modules/02-programming-in-depth/js/test/config.test.js
import { describe, it, expect, beforeEach, afterEach } from "vitest";
import { loadConfig } from "../config.js";

const ORIGINAL_ENV = process.env;

describe("config (js)", () => {
  beforeEach(() => {
    process.env = { ...ORIGINAL_ENV };
  });
  afterEach(() => {
    process.env = ORIGINAL_ENV;
  });

  it("defaults", () => {
    delete process.env.PORT;
    delete process.env.APP_ENV;
    delete process.env.LOG_LEVEL;
    const cfg = loadConfig();
    expect(cfg.port).toBe(3000);
    expect(cfg.env).toBe("dev");
    expect(cfg.logLevel).toBe("info");
  });

  it("reads env", () => {
    process.env.PORT = "5001";
    process.env.APP_ENV = "test";
    process.env.LOG_LEVEL = "debug";
    const cfg = loadConfig();
    expect(cfg.port).toBe(5001);
    expect(cfg.env).toBe("test");
    expect(cfg.logLevel).toBe("debug");
  });
});

File: modules/02-programming-in-depth/go/internal/config/config_test.go
package config

import (
	"os"
	"testing"
)

func TestDefaults(t *testing.T) {
	clearenv()
	s := Load()
	if s.Port != 3000 || s.Env != "dev" || s.LogLevel != "info" {
		t.Fatalf("unexpected defaults: %+v", s)
	}
}

func TestFromEnv(t *testing.T) {
	clearenv()
	os.Setenv("PORT", "7777")
	os.Setenv("APP_ENV", "test")
	os.Setenv("LOG_LEVEL", "debug")
	s := Load()
	if s.Port != 7777 || s.Env != "test" || s.LogLevel != "debug" {
		t.Fatalf("unexpected values: %+v", s)
	}
}

func clearenv() {
	os.Unsetenv("PORT")
	os.Unsetenv("APP_ENV")
	os.Unsetenv("LOG_LEVEL")
}
